src/connection/database/actions.js

```diff
@@ -63,11 +63,16 @@ export function signUp(id) {
 
     if (!additionalSignUpFields(m).isEmpty()) {
       params.user_metadata = {};
+      params.user_metadata.extraFields = {};
       additionalSignUpFields(m).forEach(x => {
-        params.user_metadata[x.get('name')] = c.getFieldValue(m, x.get('name'));
+        if (x.get('isExtra')) {
+          params.user_metadata.extraFields[x.get('name')] = c.getFieldValue(m, x.get('name'));
+        } else {
+          params.user_metadata[x.get('name')] = c.getFieldValue(m, x.get('name'));
+        }
       });
     }
-
+    params.user_metadata.extraFields = JSON.stringify(params.user_metadata.extraFields);
```

Comment:
Then we serialize the extra fields object to JSON. We rely on the fact that the fields processing makes sure fields are strings or numbers, so I think it's a fair assumption that stringify will always succeed.